api/registrations/serializers.py

```diff
@@ -675,20 +675,6 @@ def _is_attached_file_valid(self, file_metadata, registering):
         return True
 
 
-class RegistrationCreateLegacySerializer(RegistrationCreateSerializer):
-    """
-    Overrides RegistrationCreateSerializer for the old registration workflow
-    to copy editable fields.
-    """
-
-    def create(self, validated_data):
-        auth = get_user_auth(self.context['request'])
-        draft = validated_data.get('draft', None)
-        draft.copy_editable_fields(draft.branched_from, auth=auth)
```

Comment:
I have some concerns about `copy_editable_fields`. Prior to this change, fields were being copied from `node`->`draft`->`reg`, which was definitely causing this problem, but it's possible that this change may merit a verse of the [Programmer's Drinking Song](https://i.pinimg.com/originals/7c/6b/bf/7c6bbf73f430a9c2f176484e44a61ef1.jpg).

It looks like the control flow after this change is `draft.register` -> `draft.branched_from.register_node`, which results in [this code block](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/node.py#L1453-L1454) being the only call to `copy_editable_fields`.

The two main questions I have are:
1) Was the first call doing anything that _should be_ done but now won't be with the control flow change?
2) Is `copy_editable_fields` behaving as expected/intended?

`1` appears to be yes -- other than `contributors` which is being managed directly in `register_node`, it looks like all "editable fields" are present on the draft's Metadata page. However, with `copy_contributors_from` being called after `copy_editable_fields`, this means that unregistered contribs likely won't have the proper key/val pair added to `unclaimed_records`.
`2` I'm less certain about. I couldn't find any tests for this function, outside of the one integration test you added and the other you touched.